sdk/core/azure-core-test/src/test/java/com/azure/core/test/TestProxyTests.java

```diff
@@ -358,6 +360,56 @@ public void canGetTestProxyVersion() {
         assertNotNull(version);
     }
 
+    @Test
+    @Tag("Record")
+    public void testResetTestProxyData() throws MalformedURLException {
+        HttpURLConnectionHttpClient client = new HttpURLConnectionHttpClient();
+
+        final HttpPipeline pipeline = new HttpPipelineBuilder()
+            .httpClient(client)
+            .policies(interceptorManager.getRecordPolicy())
+            .build();
+
+        try (HttpResponse response = pipeline.sendSync(
+            new HttpRequest(HttpMethod.GET, new URL("http://localhost:3000")), Context.NONE)) {
+            assertEquals(200, response.getStatusCode());
+            HttpHeaders headers = response.getRequest().getHeaders();
+            assertNull(headers.get(HttpHeaderName.fromString("x-recording-upstream-base-uri")));
+            assertNull(headers.get(HttpHeaderName.fromString("x-recording-mode")));
+            assertNull(headers.get(HttpHeaderName.fromString("x-recording-id")));
+            assertNull(headers.get(HttpHeaderName.fromString("x-recording-skip")));
+        }
+    }
+
+    @Test
+    @Tag("Record")
+    public void testRecordWithRedirect() {
+        HttpURLConnectionHttpClient client = new HttpURLConnectionHttpClient();
+
+        HttpPipeline pipeline = new HttpPipelineBuilder()
+            .httpClient(client)
+            .policies(new RedirectPolicy(), interceptorManager.getRecordPolicy()).build();
+        URL url;
+        try {
+            url = new UrlBuilder()
+                .setHost("localhost")
+                .setPath("/getRedirect")
+                .setPort(3000)
+                .setScheme("http")
+                .toUrl();
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+
+        HttpRequest request = new HttpRequest(HttpMethod.GET, url);
+
+        try (HttpResponse response = pipeline.sendSync(request, Context.NONE)) {
+            assertEquals(200, response.getStatusCode());
+            assertEquals("http://localhost:3000/echoheaders", response.getRequest().getUrl().toString());
```

Comment:
is it correct that this test would pass even before this change because base URL is localhost for all requests?
Can we make 307 return location like "http://really-anything-but-not-localhost/foo/bar" and then check `HttpURLConnectionHttpClient` was called with it (or use another mock client or policy under recording)?

Reply:
The base url is http://localhost:3000/ but this test case particularly tests if the 200 status code on the response is returned with the expected redirected url. Plus, the recording shows the two outgoing requests currently. Should that be enough?